components/CheckDocumentation/connections/BlockingQuery.tsx

```diff
@@ -0,0 +1,129 @@
+} from "../../../util/checks";
+
+import SQL from "../../SQL";
+
+import { useSmartAnchor } from "../../SmartAnchor";
+
+const BlockingQueryTrigger: React.FunctionComponent<CheckTriggerProps> = ({
+  config,
+}) => {
+  return (
+    <>
+      <p>
+        Detects queries currently blocking more than the specified threshold of{" "}
+        <code>{config.settings["blocked_count"]}</code> queries. When the oldest
+        blocked query is running for longer than the specified threshold of{" "}
+        <code>{config.settings["warning_blocked_age_secs"]}</code> seconds,
+        creates an issue with severity "warning". Escalates to "critical" if the
+        oldest blocked query continues to run longer than{" "}
+        <code>{config.settings["critical_blocked_age_secs"]}</code> seconds.
+        Resolves automatically once these queries stop meeting the criteria.
+      </p>
+      <p>
+        Ignores any blocking queries that contain the{" "}
+        <code>/* pganalyze:no-alert */</code> or{" "}
+        <code>/* pganalyze=no-alert */</code> magic comment.
+      </p>
+    </>
+  );
+};
+
+const BlockingQueryGuidance: React.FunctionComponent<CheckGuidanceProps> = ({
+  urls: { backendsUrl, SettingLink },
+  issue,
+}) => {
+  const Link = useSmartAnchor();
+  const issueDetails = issue && JSON.parse(issue.detailsJson);
+  const showAutovacuum = issueDetails?.["has_autovacuum_to_prevent_wraparound"];
+  return (
+    <div>
+      <h4>Impact</h4>
+      <p>
+        Queries that are blocking many other queries for certain amount of time
+        likely holding some strong locks. These queries can block some trivial
+        SELECT queries sometimes, which can cause the whole system down. They
+        can also prevent VACUUM (including autovacuum) from cleaning up dead
+        rows, leading to table or index bloat. They may also be consuming a lot
+        of I/O or CPU resources, impacting overall performance.
+      </p>
+      <h4>Common Causes</h4>
+      <ul>
+        {showAutovacuum && (
+          <li>
+            <h5>Autovacuum (to prevent wraparound)</h5>
+            <p>
+              In general, autovacuum workers don't block other queries. Even if
+              the other query tries to acquire a lock that conflicts with the
+              running autovacuum, lock acquisition will interrupt the
+              autovacuum. However, if the autovacuum is running to prevent
+              transaction ID wraparound, the vacuum won't be interrupted and
+              will block other queries trying to acquire conflicting locks.
+            </p>
+          </li>
+        )}
+        <li>
+          <h5>Locks</h5>
+          <p>
+            The queries may be holding some strong locks. Check the{" "}
+            <Link to={backendsUrl}>Connection Traces</Link> page for the queries
+            these queries are blocking. You can check{" "}
+            <strong>Wait Events</strong> page for the queries waiting for these
```

Comment:
```suggestion
            <strong>Wait Events</strong> page for these
```

Reply:
Actually this will change the meaning, it's talking about checking the wait events of _blocked queries_, but removing this will become _blocking queries_. I'll tweak the whole wordings of sentence around here to make these clear.
Although, I think this whole section can use the re-write (with the title "Other long-running queries"), so I'll do some second try 👍